### PR TITLE
Add branding and version number templating support

### PR DESCRIPTION
**Pull request by huazhihao**
Not tested with iCenter yet

### CODE REVIEW

**Comment by BobBall** [comment]
Looks good.
Please generate an ICS plugin at least - I can test it if you don't have iCenter installed.

File: PluginInstaller/Create-PluginInstaller.ps1 [hunk 1]
```diff
@@ -28,8 +28,8 @@ param(
   #$lib = "$wix\\WixUI.wixlib",
   $ui_ref = "WixUI_Mondo",
   $title = "UserPlugins",
-  $manufacturer = "Citrix",
-  $description = "XenCenter Plugins",
+  $manufacturer = $env:MANUFACTURER_NAME,
+  $description = $env:PRODUCT_NAME + " Plugins",
   $product_version = "1.0.0.0",
   $upgrade_code = "8282b90a-cb51-4c02-a1c1-ecfcff9861bf",
   $product_code = "8282b90a-cb51-4c02-a1c1-ecfcff9861bf",
```
Comment (BobBall) [inline PluginInstaller/Create-PluginInstaller.ps1:35]: Is this product_code now redundant?  Should we remove it or set it to be an invalid value?

File: PluginInstaller/Create-PluginInstaller.ps1 [hunk 2]
```diff
@@ -69,7 +76,7 @@ $wxs_file = ("{0}\{1}.wxs" -f (get-location), ($out).Replace(".msi", ""))
 $default_title = "UserPlugins";
 $default_title_cab = "UserPluginscab";
 $default_manufacturer = "Default Manufacturer";
-$default_description = "XenCenter Plugins";
+$default_description = $env:PRODUCT_NAME + " Plugins";
 $default_product_version = "<?version-long>";
 $default_upgrade_code = "8282b90a-cb51-4c02-a1c1-ecfcff9861bf";
 $default_product_code = "67d68e4a-82d2-4a7d-a909-cfce92dbe71a";
```
Comment (BobBall) [inline PluginInstaller/Create-PluginInstaller.ps1:82]: Or this one? _confused_

File: PluginInstaller/Create-PluginInstaller.ps1 (hunk 1 above)
Comment (huazhihao) [inline PluginInstaller/Create-PluginInstaller.ps1:35]: The value of $default_upgrade_code is sort of like a place holder string that also shared by the wix template, and $upgrade_code is the default parameter for powershell. It is inherited from the old XenCenter plugin template and definitely looks ugly. Do you want it to be fixed in this change or we separate a new one to do a thorough refactor(there are some other places also smell)?

Comment (BobBall) [inline PluginInstaller/Create-PluginInstaller.ps1:35]: Understood.
Let's merge this and create another ticket on the backlog to tidy things up.

**Pull request merged**